test/adapters/integration.rb

```diff
@@ -178,6 +178,17 @@ def test_timeout
         end
       end
 
+      def test_timeout_before_open
+        valid_but_inexistent_host = 'http://10.255.255.254'
```

Comment:
I guess we are forced to revert to stubbing for now. @ricardovaleriano can you try that approach?